core/src/main/java/com/fincity/saas/core/service/connection/rest/RestAuthService.java

```diff
@@ -0,0 +1,110 @@
+	private static final String AUTH_TOKEN_FUNCTION_NAMESPACE = "authTokenFunctionNamespace";
+
+	private static final String ERROR_EVENT = "errorOutput";
+	private static final String AUTH_TOKEN = "accessToken";
+	private static final String EXPIRES_IN = "expiresIn";
+
+	private final CoreMessageResourceService msgService;
+	private final CoreTokenDAO coreTokenDAO;
+	private final BasicRestService basicRestService;
+	private final CoreFunctionService coreFunctionService;
+
+	public RestAuthService(CoreMessageResourceService msgService, CoreTokenDAO coreTokenDAO,
+	                       BasicRestService basicRestService, CoreFunctionService coreFunctionService) {
+
+		this.msgService = msgService;
+		this.coreTokenDAO = coreTokenDAO;
+		this.basicRestService = basicRestService;
+		this.coreFunctionService = coreFunctionService;
+
+	}
+
+	@Override
+	public Mono<RestResponse> call(Connection connection, RestRequest request) {
+
+		return FlatMapUtil.flatMapMono(
+
+						() -> {
+							Map<String, Object> connectionDetails = connection.getConnectionDetails();
+
+							String authTokenFunctionName = (String) connectionDetails.get(AUTH_TOKEN_FUNCTION_NAME);
+							String authTokenFunctionNameSpace = (String) connectionDetails.get(AUTH_TOKEN_FUNCTION_NAMESPACE);
+
+							return coreFunctionService.execute(authTokenFunctionNameSpace, authTokenFunctionName,
+									connection.getAppCode(), connection.getClientCode(), null, null);
+						},
+
+						authTokenOutput -> Mono.just(authTokenOutput.allResults().get(0)),
+
+						(authTokenOutput, outputResult) -> {
+
+							if (outputResult.getName().equals(ERROR_EVENT)) {
+								return msgService.throwMessage(msg -> new GenericException(HttpStatus.BAD_REQUEST, msg),
+										CoreMessageResourceService.NOT_ABLE_TO_CREATE_TOKEN, connection.getName(), outputResult);
+							}
+
+							Map<String, JsonElement> eventMap = outputResult.getResult();
+							String authToken = eventMap.get(AUTH_TOKEN).getAsString();
+							long expiresIn = eventMap.get(EXPIRES_IN).getAsLong();
+
+							return coreTokenDAO.create(new CoreToken()
+									.setClientCode(connection.getClientCode())
+									.setAppCode(connection.getAppCode())
+									.setConnectionName(connection.getName())
+									.setTokenType(CoreTokensTokenType.ACCESS)
+									.setToken(authToken)
+									.setIsRevoked(Boolean.FALSE)
+									.setExpiresAt(LocalDateTime.now().plusSeconds(expiresIn)));
+						},
+						(authTokenOutput, outputResult, coreToken) ->
+								makeRestCall(connection, request, coreToken.getToken()))
+				.contextWrite(Context.of(LogUtil.METHOD_NAME, "RestAuthService.call"));
+	}
+
+	private Mono<RestResponse> makeRestCall(Connection connection, RestRequest request, String accessToken) {
+
+		Object tokenPrefix = connection.getConnectionDetails().get("headerPrefix");
+
+		String authorizationHeader = (tokenPrefix != null) ? tokenPrefix + " " + accessToken : "Bearer " + accessToken;
```

Comment:
Please don't add any prefix here if none is mentioned. Bearer is not mandatory in many services.